packages/vats/src/proposals/upgrade-asset-reserve-proposal.js

```diff
@@ -0,0 +1,90 @@
+import { E } from '@endo/far';
+import { deeplyFulfilled } from '@endo/marshal';
+import { makeTracer } from '@agoric/internal';
+
+const tracer = makeTracer('UpgradeAssetReserve');
+
+/**
+ * @param {BootstrapPowers & {
+ *   consume: {
+ *     economicCommitteeCreatorFacet: any;
+ *     reserveKit: any;
+ *   };
+ *   produce: {
+ *     reserveKit: any;
+ *   };
+ * }} powers
+ * @param {object} options
+ * @param {{ assetReserveRef: VatSourceRef }} options.options
+ */
+export const upgradeAssetReserve = async (
+  {
+    consume: {
+      economicCommitteeCreatorFacet: electorateCreatorFacet,
+      reserveKit: reserveKitP,
+      instancePrivateArgs: instancePrivateArgsP,
+    },
+    produce: { reserveKit: reserveKitWriter },
+  },
+  options,
+) => {
+  const { assetReserveRef } = options.options;
+
+  assert(assetReserveRef.bundleID);
+  tracer(`ASSET RESERBE BUNDLE ID: `, assetReserveRef);
+
+  const [reserveKit, instancePrivateArgs] = await Promise.all([
+    reserveKitP,
+    instancePrivateArgsP,
+  ]);
+  const { governorCreatorFacet, instance } = reserveKit;
+
+  const [originalPrivateArgs, poserInvitation] = await Promise.all([
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore Local tsc sees this as an error but typedoc does not
+    deeplyFulfilled(instancePrivateArgs.get(instance)),
+    E(electorateCreatorFacet).getPoserInvitation(),
+  ]);
+
+  const newPrivateArgs = harden({
+    ...originalPrivateArgs,
+    initialPoserInvitation: poserInvitation,
+  });
+
+  const adminFacet = await E(governorCreatorFacet).getAdminFacet();
+
+  reserveKitWriter.reset();
+  reserveKitWriter.resolve(
+    harden({
+      ...reserveKit,
+      adminFacet,
+    }),
+  );
```

Comment:
As long as you're doing this, might as well clean up [the original](https://github.com/Agoric/agoric-sdk/blob/4feddb0e0fbdf2343622c7424c6e6d3de013c85d/packages/inter-protocol/src/proposals/econ-behaviors.js#L172), (we want those proposals to be usable to start a new chain, or to copy, even if we won't run them again on mainNet) and add a comment here as to why this fix is necessary.